QUARKUS-2830.md

```diff
@@ -0,0 +1,50 @@
+# QUARKUS-2829 - Full Support Micrometer metrics also when using managed Vert.x in Quarku
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-2830
+
+As part of supporting Vert.x use-cases on Quarkus we need to support collecting metrics using micrometer for the managed Vert.x instance in Quarkus. This is also a requirement for some layered products like AMQ Streams and OpenShift Serverless that today are using Vert.x, but needs to move to use Quarkus.
+
+## Scope  of the implementation
+Managed Vert.x in Quarkus contains following extensions:
+- Messaging clients:
+  AMQP,Kafka, MQTT. Extension names: quarkus-smallrye-reactive-messaging-amqp, *-kafka, *-mqtt.
+- DB clients: Mssql, DB2, Oracle, MySQL, Postgres(pg). Extension names: quarkus-reactive-(name)-client.
+- Redis client: quarkus-redis-client
+- quarkus-mailer
+- quarkus-vertx (the core API)
+
+In comparison to internal implementation (QUARKUS-2829), the following additions were made:
+- mailer metrics
+- grpc metrics
+- database metrics (db2, mssql, mysql, oracle, postgresql, redis)
+- TCP metrics (bytes_written, bytes_read, connections)
+- UDP metrics (bytes_written, bytes_read, errors)
+- EventBus metrics (eventBus.published, eventBus.sent, replyFailures, handlers, delivered, discarded)
+- Http client metrics (queue.delay, queue.size)
+
+## Scope of testing
+Cover, that metrics are being created properly for applications, running on different targets (baremetal, openshift and serverless).
+
+### Existing tests
+Upstream tests contain coverage for Redis, HTTP client, telnet, eventbus, TCP and UDP metrics on baremetal.
+[Quarkus start-stop test suite](https://github.com/quarkus-qe/quarkus-startstop) contains several modules, which contain Vertx.X based extensions, the most interesting for our current efforts are monitoring-micrometer-prometheus-kafka-reactive which cover gathering of Micrometer metrics from AMQ Streams applications.
+As a part of QUARKUS-2829, we added checks for https server metrics on baremetal, openshift and serverless.
+
+## Automated test development
+We should add checks:
+- for databases properties, either in vertx-sql module, or add database usage into vertx-http module.
+- for event-bus specific properties on different targets
+
```

Comment:
Do you have plans to cover grpc metrics too? Where would be the coverage located?

Reply:
Yes, that is a good idea. Added a plan for grpc.